### PR TITLE
Superpowered v2.6.7

### PR DESCRIPTION
**Pull request by dodds-cc**
Release v2.6.7, with the following changes

1. Superpowered Advanced Audio Player:

- Song duration is returned as a float value
- Current pitch bend percent can be retrieved

2. Added Advanced Audio Player pitch bend example to the Javascript repository.

### CODE REVIEW

File: examples/example_pitchbend/main.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+var content = null;      // The <div> displaying everything.
+var pitchShift = 0;      // The current pitch shift value.
+var currentPath = null;
+var pbPerc = null;
+
+function changePitchShift(e) {
+    // limiting the new pitch shift value
+    let value = parseInt(e.target.value);
+    pitchShift += value;
+    if (pitchShift < -12) pitchShift = -12; else if (pitchShift > 12) pitchShift = 12;
```
Comment (gaborszanto) [inline examples/example_pitchbend/main.js:15]: Can we have more "Javascript" here? Such as:

`pitchShift = Math.min(12, Math.max(-12, pitchShift + parseInt(e.target.value)));`

File: examples/example_pitchbend/main.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    // displaying the value
+    document.getElementById('pitch-shift-display').innerText = ' pitch shift: ' + ((pitchShift < 1) ? pitchShift : '+' + pitchShift) + ' ';
```
Comment (gaborszanto) [inline examples/example_pitchbend/main.js:17]: I think it's "textContent" these days, not "innerText".

File: examples/example_pitchbend/main.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    let faster = 1;
+    const value = e.target.value;
+    if (value < 0) {
+        faster = 0;
+    }
+    audioNode.sendMessageToAudioScope({
+        'pitchBend': true,
+        maxPercent: Math.abs(value/100),
+        bendStretch: 0,
+        faster,
```
Comment (gaborszanto) [inline examples/example_pitchbend/main.js:44]: Isn't this a bug? I think it's missing the key "faster".

Something like: `faster: value < 0 ? 0 : 1,`

File: examples/example_pitchbend/processor.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            bendStretch: message.bendStretch,
+            faster: message.faster,
+            holdMs: message.holdMs
+        } : undefined;
+    }
+
+    processAudio(inputBuffer, outputBuffer, buffersize, parameters) {
+        if (this.pitchBend) {
+            this.player.pitchBend(this.pitchBend.maxPercent, this.pitchBend.bendStretch, this.pitchBend.faster, this.pitchBend.holdMs);
+            if (this.cancelledPitchBend) this.cancelledPitchBend = false;
```
Comment (gaborszanto) [inline examples/example_pitchbend/processor.js:36]: Is the "if" necessary here? Just set it to false?

**Review by gaborszanto — CHANGES_REQUESTED** [review]
Please check my comments.

File: examples/example_pitchbend/assets/processor.js [hunk 5]
```diff
@@ -33,11 +31,10 @@ class MyProcessor extends SuperpoweredWebAudio.AudioWorkletProcessor {
     processAudio(inputBuffer, outputBuffer, buffersize, parameters) {
         if (this.pitchBend) {
             this.player.pitchBend(this.pitchBend.maxPercent, this.pitchBend.bendStretch, this.pitchBend.faster, this.pitchBend.holdMs);
-            if (this.cancelledPitchBend) this.cancelledPitchBend = false;
         } else if (!this.cancelledPitchBend) {
             this.player.endContinuousPitchBend();
-            this.cancelledPitchBend = true;
         }
+        this.cancelledPitchBend = !this.cancelledPitchBend;
```
[the remark this answers is not in the document]
Comment (dodds-cc) [reply examples/example_pitchbend/assets/processor.js]: yes, I agree. Updated.

**Comment by dodds-cc** [comment]
@gaborszanto Can you merge this now please?

**Review by gaborszanto — APPROVED** [review]

**Pull request merged**